pkg/api/frontend_logging_test.go

```diff
@@ -0,0 +1,149 @@
+package api
+
+import (
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/getsentry/sentry-go"
+	"github.com/grafana/grafana/pkg/models"
+	log "github.com/inconshreveable/log15"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type logScenarioFunc func(c *scenarioContext, logs []*log.Record)
+
+func logSentryEventScenario(t *testing.T, desc string, event frontendSentryEvent, fn logScenarioFunc) {
+	t.Run(desc, func(t *testing.T) {
+		logs := []*log.Record{}
+		origHandler := frontendLogger.GetHandler()
+		frontendLogger.SetHandler(log.FuncHandler(func(r *log.Record) error {
+			logs = append(logs, r)
+			return nil
+		}))
+		t.Cleanup(func() {
+			frontendLogger.SetHandler(origHandler)
+		})
+
+		sc := setupScenarioContext("/log")
+		hs := HTTPServer{}
+
+		sc.defaultHandler = Wrap(func(w http.ResponseWriter, c *models.ReqContext) Response {
```

Comment:
Is there a reason not to use sc.handlerFunc instead? That hook is called by sc.defaultHandler if I'm not mistaken.

Reply:
I reused this piece of code from another test without without polishing it too much. But now that I look at it, don't need to set either `handlerFunc` or `defaultHandler` here, as not using any of the helper scenarios that actually use these. Updated code to reflect it

Reply:
Great :)